src/js/components/homepage/visualizations/treemap/BudgetFunctions/BudgetFunctionCell.jsx

```diff
@@ -88,89 +99,65 @@ export default class TreeMapCell extends React.Component {
-                    this.mouseIn(this.props.label, this.props.value, "#F2B733");
+                onMouseEnter={() => {
+                    this.props.toggleTooltipIn(this.props.functionID);
                 }}
                 onMouseLeave={() => {
-                    this.mouseIn('none', '', this.props.color);
+                    this.props.toggleTooltipOut();
+                }}
+                onClick={() => {
+                    if (this.props.clickable) {
+                        this.props.toggleSubfunction(this.props.functionID);
+                    }
                 }}>
                 <rect
                     className="tile"
-                    width={width}
-                    height={height}
+                    width={this.props.width}
+                    height={this.props.height}
                     style={{
-                        fill: this.state.color,
-                        stroke: this.toggleBorders()[0],
-                        strokeOpacity: this.toggleBorders()[1],
+                        fill: this.props.color,
+                        stroke: this.props.strokeColor,
+                        strokeOpacity: this.props.strokeOpacity,
                         strokeWidth: "2px",
                         padding: "10px"
                     }} />
                 <text
-                    className="category"
-                    x={(width / 2)}
-                    y={height / 2}
-                    width={width}
+                    className={`category ${this.props.textClass}`}
+                    x={(this.props.width / 2)}
+                    y={this.props.height / 2}
+                    width={this.props.width}
                     textAnchor="middle"
                     ref={(text) => {
                         this.svgText = text;
                     }}
                     style={{
-                        display: labelView
+                        display: this.props.labelView,
+                        fill: this.props.textColor,
+                        textShadow: this.props.textShadow
                     }}>
                     {this.state.label}
                 </text>
                 <text
-                    className="value"
-                    x={(width / 2) - 2}
-                    y={(height / 2) + 20}
-                    width={width}
+                    className={`value ${this.props.textClass}`}
+                    x={(this.props.width / 2) - 2}
+                    y={(this.props.height / 2) + 20}
+                    width={this.props.width}
                     textAnchor="middle"
                     style={{
-                        display: percentView
+                        display: this.props.percentView,
+                        fill: this.props.textColor,
+                        textShadow: this.props.textShadow
                     }}>
                     {parseFloat((this.props.value / this.props.total) * 100).toFixed(1)}%
```

Comment:
If we're going to add special treemap functions to MoneyFormatter we may as well use them.....